torch/csrc/api/include/torch/nn/modules/rnn.h

```diff
@@ -1,121 +1,113 @@
 #pragma once
 
-#include <torch/detail.h>
 #include <torch/nn/module.h>
 
 #include <torch/csrc/autograd/variable.h>
 
 #include <ATen/Error.h>
+#include <ATen/optional.h>
 
 #include <cstdint>
+#include <functional>
 #include <memory>
 #include <vector>
 
+namespace torch { namespace nn {
+class Dropout;
+}} // namespace torch::nn
+
 namespace torch { namespace nn {
 template <typename Derived>
 class RNNBase : public CloneableModule<Derived> {
  public:
-  // These must line up with the CUDNN mode codes
-  enum RNNMode { RNN_RELU = 0, RNN_TANH = 1, LSTM = 2, GRU = 3 };
+  // These must line up with the CUDNN mode codes:
+  // https://docs.nvidia.com/deeplearning/sdk/cudnn-developer-guide/index.html#cudnnRNNMode_t
+  enum class CuDNNMode { RNN_RELU, RNN_TANH, LSTM, GRU };
```

Comment:
I feel like it's better to keep the explicit setting of = 0, 1, 2, 3,  otherwise code might not work on a different compiler that sets it to a different value for whatever stupid reason.

Reply:
I think it's guaranteed to be this range

Reply:
Yes, but it does look a bit weird to state that the enum must line up with something and then not specifying the actual values. Explicit enum values would also protect against detrimental future modifications.